tests/python/pants_test/backend/python/tasks/native/test_ctypes_integration.py

```diff
@@ -267,3 +270,55 @@ def test_native_compiler_option_sets_integration(self, toolchain_variant):
     })
     self.assert_success(pants_run)
     self.assertIn('x=3, f(x)=12600000', pants_run.stdout_data)
+
+  _zero_out_module_contents = dedent("""\
+    op lib: ['LIB_HANDLE', 'OP_LIST', '_InitOpDefLibrary', '__builtins__', '__doc__', '__loader__', '__name__', '__package__', '__spec__', '_collections', '_common_shapes', '_context', '_core', '_dispatch', '_dtypes', '_errors', '_execute', '_op_def_lib', '_op_def_library', '_op_def_pb2', '_op_def_registry', '_ops', '_pywrap_tensorflow', '_six', '_tensor_shape', 'deprecated_endpoints', 'tf_export', 'zero_out', 'zero_out_eager_fallback']
+    """)
+
+  def _assert_tensorflow_binary_creation(self, target):
+    with temporary_dir() as tmp_dir:
+      self.do_command(
+        'binary',
+        target,
+        config={
+          'native-build-step': {
+            'toolchain_variant': 'llvm',
+          },
+          GLOBAL_SCOPE_CONFIG_SECTION: {
+            'pants_distdir': tmp_dir,
+          },
+        })
+      output_file = os.path.join(tmp_dir, 'zero-out-custom-op-hello-world.pex')
+      return output_file
+
+  def _assert_tensorflow_run(self, target):
+    stdout = self.do_command(
+      '-q',
+      'run',
+      target,
+      config={
+        'native-build-step': {
+          'toolchain_variant': 'llvm',
+        },
+      }).stdout_data
+    self.assertEqual(stdout.decode('utf-8'), self._zero_out_module_contents)
+
+  def test_tensorflow_cpu_binary_and_run(self):
+    pex_file = self._assert_tensorflow_binary_creation(self._tensorflow_cpu_target)
+    self.assertEqual(
+        subprocess.run([pex_file], check=True, capture_output=True).stdout,
```

Comment:
Sadly, `capture_output` wasn't added till Py37. You'll have to use `stdout=subprocess.PIPE`.

Reply:
You'll also likely want to use `encoding='utf-8` so that it returns unicode `str`.